var/spack/repos/builtin/packages/py-ipykernel/package.py

```diff
@@ -32,19 +34,24 @@ class PyIpykernel(PythonPackage):
     depends_on('python@3.5:', when='@5.2:', type=('build', 'run'))
     depends_on('python@3.7:', when='@6.0:', type=('build', 'run'))
     depends_on('py-setuptools', type='build', when='@5:')
-    depends_on('py-importlib-metadata@:4', when='@6:^python@:3.7', type=('build', 'run'))
-    depends_on('py-argcomplete@1.12.3:', when='@6:^python@:3.7', type=('build', 'run'))
+    depends_on('py-ipython-genutils', when='@5.5.6', type=('build', 'run'))
```

Comment:
Did you mean to say 5.5.5 here? Just hit this bug.

It's actually unclear to me when this dependency was added. It's always been in the imports, but wasn't documented for a long time. Did it move from the main library to a separate library at some point?

Reply:
I have added it as it is defined in setup.py. `py-ipython-genutils` is only present for [version 5.5.6](https://github.com/ipython/ipykernel/blob/5.5.6/setup.py#L88) but not for [version 5.5.5](https://github.com/ipython/ipykernel/blob/5.5.5/setup.py#L87) and is not there anymore for [version 6.0.0](https://github.com/ipython/ipykernel/blob/6.0.0/setup.py#L76)

As it seems `py-ipython-genutils` was a dependency of `py-traitlets` before as well but with the new release it was removed there (I opened a [PR for this release](https://github.com/spack/spack/pull/27322) last week) and thus exposing this bug in py-ipykernel (see [here](https://github.com/ipython/ipykernel/issues/759) and [here](https://github.com/ipython/ipykernel/issues/755))

Reply:
It seems that the ipykernel people added the [dependency after the bug reports](https://github.com/ipython/ipykernel/pull/756) and released version 5.5.6. Afterwards [dropped the requirement](https://github.com/ipython/ipykernel/pull/792) again in favor of using traitlets.